### ISSUE TITLE
[Feature Request] Pricelist Rules: Only Override Base Pricelist Rule with Matching Quantity

### ISSUE THREAD

**Issue by AaronBeaudoin** [odoo/odoo#105505]
1. Suppose I have a "base" pricelist with rules like this:

```
- Product A: $50
- Product A, Min. Quantity 10: $40
```

2. Suppose I have an "inheriting" pricelist like this:

```
- Product A: $45
- All Other Products: Use "base" pricelist.
```

- If I create an order using the "base" pricelist with 5 of Product A, I'll get the unit price of $50.
- If I create an order using the "base" pricelist with 10 of Product A, I'll get the unit price of $40.
- If I create an order using the "inheriting" pricelist with 5 of Product A, I'll get the unit price of $45.
- If I create an order using the "inheriting" pricelist with 10 of Product A, I'll **still** get the unit price of $45.  
  **This is not what I want.** I want the "base" pricelist's minimum quantity unit price of $40.

I would like to be able to "continue to use the current pricelist's base pricelist minimum quantity rules".

Using a "base" pricelist and inheriting it is a great way to prevent repetition when providing customers with some negotiated rates but falling back to some default pricelist when necessary. However, when the base pricelist also has minimum quantity "bulk" rules for a product, they seem to be ignored if the "inheriting" pricelist supplies **any rules at all** for that product.

It makes more sense to only "override" rules on the "inheriting" pricelist if the new rule matches both the product **and** the minimum quantity. That way, with the same pricelists above, I'd get the behavior I want. However, if I did want the same behavior as currently, I would simply a `Product A, Min Quantity 10: $45` line on the "inheriting" pricelist.

**Comment by kup-odoo** [comment]
Hello @AaronBeaudoin ,
Appreciate your request.

As we only discussing issues here, you can post your bugs/issues if any.
For feature request, you can submit support ticket via odoo.com/help

please have a look if we can close this from issues.
Thank You

**Comment by AaronBeaudoin** [comment]
I have submitted a support ticket, but I think Odoo should have a **public** place to create feature requests, so that everyone can see the amount of interest in a given ticket via upvotes, and so that there is more accountability and visibility in the case that feature requests get buried.

**Comment by Feyensv** [comment]
What you're requesting here is way too magic and would be confusing to the users.

Only one rule is applied at a time, and the ordering to know the rule to apply is constant.
Targeted rules (not global ones applied on all products) will always take precedence over global ones, which means the rule `- Product A: $45` will always be applied for product A, and not the rule `- All Other Products: Use "base" pricelist.`.

**Comment by AaronBeaudoin** [comment]
I understand where you're coming from, but the feature I'm requesting fulfills a very real-world use case that there is currently no way to accomplish.

One of the main advantages of the current "base pricelist" rule feature is the ability to give a customer some custom rates on products but then "fallback" to some base prices for all other products, **avoiding the need to repeat the rules for all those other products on every single pricelist**.

However, it is also common to need to give **bulk pricing** to all customers under all circumstances, even when giving the customer some custom rates (via their own pricelist). In this case, the only way to do this currently is to **redefine** the same "bulk pricing" (`min_quantity > 0`) rules on the customer's "inheriting" pricelist. In the case where you have 200+ pricelists for different customers (which describes our company), you now have to update all 200 each time you adjust bulk pricing, which is needlessly repetitive, prone to error, and sidesteps what I would consider to be the purpose of the current "base pricelist" rule feature described above.

So, in short, while "base pricelist" rules have given the ability to avoid repetition while `min_quantity` is zero, as soon as we need to also do bulk pricing, we have to add any `min_quantity > 1` rules to **all** pricelists where the product's "base" price is adjusted, partially defeating the whole purpose of "inheriting" pricelists.

**Comment by Feyensv** [comment]
Both cases exists, but we cannot satisfy both solution together (or without adding complexity, confusing basic users).
We consider the current supported use case more frequent and we won't change that for now.
